src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2121,7 +2121,7 @@ void SharedRuntime::monitor_enter_helper(oopDesc* obj, BasicLock* lock, JavaThre
   if (PrintBiasedLockingStatistics) {
     Atomic::inc(BiasedLocking::slow_path_entry_count_addr());
   }
-  Handle h_obj(THREAD, obj);
+  Handle h_obj(thread, obj);
```

Comment:
Can you make this one 'current'?

Reply:
Unfortunately not - the JRT_BLOCK_NO_ASYNC further down requires "thread".

Reply:
Then keep it THREAD.  That implies current thread.